pkgs/build-support/bintools-wrapper/default.nix

```
@@ -56,6 +56,8 @@ let
     else if targetPlatform.libc == "nblibc"           then "${libc_lib}/libexec/ld.elf_so"
     else if targetPlatform.system == "i686-linux"     then "${libc_lib}/lib/ld-linux.so.2"
     else if targetPlatform.system == "x86_64-linux"   then "${libc_lib}/lib/ld-linux-x86-64.so.2"
+    # check before generic arm condition
+    else if targetPlatform.isRemarkable               then "/lib/ld-linux-armhf.so.3"
     # ARM with a wildcard, which can be "" or "-armhf".
     else if (with targetPlatform; isAarch32 && isLinux)   then "${libc_lib}/lib/ld-linux*.so.3"
```

I think the second condition already will work fine
```suggestion
    # ARM with a wildcard, which can be "" or "-armhf".
    else if (with targetPlatform; isAarch32 && isLinux)   then "${libc_lib}/lib/ld-linux*.so.3"
```

---

@Ericson2314 As per the earlier discussion, I think we do need something in here to allow using the `/lib/ld-linux-armhf.so.3` linker to allow producing standalone binaries that run on the device without a Nix installation, which I think is the main use case for this toolchain. This is analogous to the entry just above, for bionic (`else if targetPlatform.libc == "bionic"           then "/system/bin/linker"`)

---

@pl-semiotics I'm a bit dubious of supporting the impure/prebuilt ways if the prebuilt stuff works. (I'd like to get rid of the prebuilt android stuff too, see https://github.com/NixOS/nixpkgs/issues/38411.)

As I see it, there are use-case Nix should support, if they are possible:
 - Build from scratch, statically linked, should be able to drop in
 - Build from scratch, dynamically linked, can use `patchelf` to replace libraries with pre-installed versions if desired.